deploy/operatorsource.crd.yaml

```diff
@@ -2,12 +2,65 @@ apiVersion: apiextensions.k8s.io/v1beta1
 kind: CustomResourceDefinition
 metadata:
   name: operatorsources.marketplace.redhat.com
+  annotations:
+    displayName: Operator Source
+    description: Represents an OperatorSource.
 spec:
   group: marketplace.redhat.com
   names:
     kind: OperatorSource
     listKind: OperatorSourceList
     plural: operatorsources
     singular: operatorsource
+    shortNames:
+    - os
   scope: Namespaced
   version: v1alpha1
+  additionalPrinterColumns:
+  - name: Type
+    type: string
+    description: The type of the OperatorSource
+    JSONPath: .spec.type
+  - name: Endpoint
+    type: string
+    description: The endpoint of the OperatorSource
+    JSONPath: .spec.endpoint
+  - name: Registry
+    type: string
+    description: App registry namespace
+    JSONPath: .spec.registryNamespace
+  - name: Status
+    type: string
+    description: Current status of the OperatorSource
+    JSONPath: .status.currentPhase.phase.name
+  - name: Message
+    type: string
+    description: Message associated with the current status
+    JSONPath: .status.currentPhase.phase.message
+  - name: Age
+    type: date
+    JSONPath: .metadata.creationTimestamp
+  validation:
+    openAPIV3Schema:
+      properties:
+        spec:
+          type: object
+          description: Represents the configuration of an OperatorSource
+          required:
+          - spec
+          type: object
+          description: Spec for an OperatorSource.
+          required:
+          - type
+          - endpoint
+          - registryNamespace
+          properties:
+            type:
+              type: string
```

Comment:
Can you specify a predefined set of values here?

Reply:
Not that I know of for strings. What predefined set of values would you want here?

Reply:
I have addressed using a pattern which is just `appregistry` for now. Please review.